src/charm.py

```diff
@@ -284,22 +318,22 @@ def _get_saml_config(self) -> typing.Dict[str, typing.Any]:
         Returns:
             Dictionary with the SAML configuration settings..
         """
-        ubuntu_one_fingerprint = "32:15:20:9F:A4:3C:8E:3E:8E:47:72:62:9A:86:8D:0E:E6:CF:45:D5"
-        ubuntu_one_staging_fingerprint = (
-            "D2:B4:86:49:1B:AC:29:F6:A4:C8:CF:0D:3A:8F:AD:86:36:0A:77:C0"
-        )
-        saml_fingerprints = {
-            "https://login.ubuntu.com/+saml": ubuntu_one_fingerprint,
-            "https://login.staging.ubuntu.com/+saml": ubuntu_one_staging_fingerprint,
-        }
         saml_config = {}
 
-        if self.config.get("saml_target_url"):
-            saml_config["DISCOURSE_SAML_TARGET_URL"] = self.config["saml_target_url"]
+        relation = self.model.get_relation(DEFAULT_RELATION_NAME)
+        if (
+            relation is not None
+            and relation.data[self.app]
+            and relation.app
+            and relation.data[relation.app]
+        ):
+            saml_config["DISCOURSE_SAML_TARGET_URL"] = relation.data[relation.app][
+                "single_sign_on_service_redirect_url"
+            ]
```

Comment:
Similar to the comment below, I think we can improve this part (accessing the relation data directly) if the SAML charm library can provide the `endpoints` information outside of the event handler.